frontend/src/index.tsx

```diff
@@ -1,15 +1,44 @@
-import React from 'react';
+import React, { lazy, Suspense } from 'react';
 import ReactDOM from 'react-dom/client';
+import {
+  createBrowserRouter,
+  Route,
+  RouterProvider,
+  useParams,
+} from 'react-router-dom';
 import './index.css';
 import App from './App';
 import reportWebVitals from './reportWebVitals';
 
+const Loader = () => {
+  const { gameName } = useParams();
+
+  const GamePage = lazy(() => import(`./games/${gameName}`));
+
+  return (
+    <Suspense fallback={<div>Loading...</div>}>
+      <GamePage />
+    </Suspense>
+  );
+};
+
+const router = createBrowserRouter([
+  {
+    path: '/',
+    element: <App />,
+  },
+  {
+    path: '/games/:gameName',
+    element: <Loader />,
+  },
+]);
+
 const root = ReactDOM.createRoot(
   document.getElementById('root') as HTMLElement,
 );
 root.render(
   <React.StrictMode>
-    <App />
+    <RouterProvider router={router} />
```

Comment:
Jeigu neklystu @Dax0s RouterProvider turėtų eiti tarp app, o index turėtų būti gana clean. But if it works it works. Convention gerai nežinau.

Reply:
Yep, man irgi taip atrodo, routerio dalykai turetu buti App.tsx